components/suggest/src/rs.rs

```diff
@@ -100,30 +100,44 @@ pub(crate) enum SuggestRecord {
     GlobalConfig(DownloadedGlobalConfig),
 }
 
+/// Implemented by the various attachment types
+pub trait SuggestAttachment: DeserializeOwned + std::fmt::Debug {
+    /// Get fields that we saw when ingesting this attachment, but don't recognize
+    fn unknown_fields(&self) -> &serde_json::Map<String, serde_json::Value>;
+
+    fn has_unknown_fields(&self) -> bool {
+        !self.unknown_fields().is_empty()
+    }
+}
+
 /// Represents either a single value, or a list of values. This is used to
 /// deserialize downloaded attachments.
 #[derive(Clone, Debug, Deserialize)]
 #[serde(untagged)]
-enum OneOrMany<T> {
+pub enum OneOrMany<T> {
     One(T),
     Many(Vec<T>),
 }
 
-/// A downloaded Remote Settings attachment that contains suggestions.
-#[derive(Clone, Debug, Deserialize)]
-#[serde(transparent)]
-pub(crate) struct SuggestAttachment<T>(OneOrMany<T>);
-
-impl<T> SuggestAttachment<T> {
+impl<T> OneOrMany<T> {
     /// Returns a slice of suggestions to ingest from the downloaded attachment.
-    pub fn suggestions(&self) -> &[T] {
-        match &self.0 {
+    pub fn as_slice(&self) -> &[T] {
+        match &self {
             OneOrMany::One(value) => std::slice::from_ref(value),
             OneOrMany::Many(values) => values,
         }
     }
 }
 
+impl<T: SuggestAttachment> OneOrMany<T> {
+    /// Did any of the attachments have an unknown field?
```

Comment:
Nit: as discussed above, `OneOrMany` would only involve one attachment, so maybe `Did any of the payloads (i.e. JSON values) in this attachment have an unknown field?`.